### PR TITLE
Convert action to composite

### PR DESCRIPTION
**Pull request by knightdave**
Convert to composite action. Close https://github.com/insightsengineering/thevalidatoR/issues/29.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -19,12 +19,71 @@ inputs:
...
+      run: echo "::set-output name=R_VERSION::$(R --version | head -1 | awk '{print $3}')"
+      shell: bash
 
-branding: # https://feathericons.com/
-  icon: 'award'
-  color: 'blue'
+    - name: Set R Library home on Linux
+      
+      run: |
+        mkdir -p /__w/_temp/Library
```
Comment (dinakar29) [inline action.yml:45]: Isn't this what Github Action uses on the runner's host machine? I'd be wary of putting anything there an using `$GITHUB_WORKSPACE` instead

File: action.yml [hunk 2]
```diff
@@ -19,12 +19,71 @@ inputs:
...
+      if: "contains(inputs.no_cache, 'false')"
+      id: texlive_version
+      run: |
+        echo "TEXMFHOME=/__w/_temp/texlive/texmf-home" >> $GITHUB_ENV
+        echo "TEXMFVAR=/__w/_temp/texlive/texmf-var" >> $GITHUB_ENV
+        export TEXMFHOME=/__w/_temp/texlive/texmf-home
+        export TEXMFVAR=/__w/_temp/texlive/texmf-var
+        tlmgr init-usertree
+        echo "::set-output name=TEX_LIVE_VERSION::$(tlmgr --version |tail -1 |awk '{print $NF}')"
+      shell: bash
```
Comment (dinakar29) [inline action.yml:59]: Might be good to install TinyTex instead of relying on a TexLive installation.

File: action.yml (hunk 1 above)
Comment (knightdave) [reply action.yml:45]: It's because of this how action run inside docker container. It [spawns container with](https://github.com/insightsengineering/thevalidatoR/runs/4311717111?check_suite_focus=true#step:2:63):

```
 /usr/bin/docker create --name b8ad9de4427c4411bb5cb3bf84e7cbcc_rockerverse411_9885f6 --label e28490 --workdir /__w/thevalidatoR/thevalidatoR --network github_network_b75ab2503ef34c109dd2b5f9d9449583  -e "HOME=/github/home" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work":"/__w" -v "/home/runner/runners/2.284.0/externals":"/__e":ro -v "/home/runner/work/_temp":"/__w/_temp" -v "/home/runner/work/_actions":"/__w/_actions" -v "/opt/hostedtoolcache":"/__t" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" --entrypoint "tail" rocker/verse:4.1.1 "-f" "/dev/null"
```

It means inside container we have this `/home/runner/work/_temp` mounted as `/__w/_temp`.
`run` step is running inside container so we need to rely on this weird path. The variable `GITHUB_WORKSPACE` refers to the working path in machine - so it will be `/home/runner/work/org/repo` [doc](https://docs.github.com/en/actions/learn-github-actions/environment-variables). The path `/home/runner/work/_temp` will be the path that we want to cache with cache action.

Basically it's a workaround to properly use `_temp` directory for caching inside docker container - I can search for article where I found this.

File: action.yml (hunk 2 above)
Comment (knightdave) [reply action.yml:54]: done

File: action.yml (hunk 1 above)
Comment (knightdave) [reply action.yml]: I updated this to at least use env vars where I can

Comment (dinakar29) [reply action.yml]: Ah good. Thank you!

File: dependencies.R [hunk 3]
```diff
@@ -11,7 +11,33 @@ github_packages <- c(
...
+if(!dir.exists(paste(Sys.getenv("RUNNER_TEMP"), "TinyTeX", sep="/"))) {
+# nolint start
+  tinytex_installer <- '
+export TINYTEX_DIR=${RUNNER_TEMP}/TinyTeX
+wget -qO- "https://raw.githubusercontent.com/yihui/tinytex/master/tools/install-unx.sh" | sh -s - --admin --no-path
+mkdir -p ${RUNNER_TEMP}/TinyTeX
+cp -r ~/.TinyTeX/. ${RUNNER_TEMP}/TinyTeX
+rm -rf ~/.TinyTeX
+${RUNNER_TEMP}/TinyTeX/bin/*/tlmgr path add
+tlmgr install latex-bin luatex xetex ae bibtex context inconsolata listings makeindex metafont mfware parskip pdfcrop tex tools url xkeyval courier ec
```
Comment (dinakar29) [inline dependencies.R:29]: Ideally I'd like this to be an input to the action - i.e. add additional TeX packages if they are missing or needed by a package. 
Let's keep these packages as defaults, but let's also provide the ability to add extra pacakges.

Comment (knightdave) [reply dependencies.R]: the whole list without `courier` and `ec` comes from rocker/verse - they are installed by default.
I added parameter that allows install more of them then only those two.

**Comment by knightdave** [comment]
Finally tmgr caching working:

No cache:
Screenshot 2021-11-24 at 16 31 49


With cache:
Screenshot 2021-11-24 at 16 37 22

(aragog 7m --> 2m decreased time)

**Comment by knightdave** [comment]
@dinakar29  - do you want me to add something more? Is it OK to merge?

**Comment by dinakar29** [comment]
> @dinakar29 - do you want me to add something more? Is it OK to merge?

Can we add this? https://github.com/insightsengineering/thevalidatoR/pull/45/files#r756155629

**Comment by knightdave** [comment]
> Can we add this? [#45 (comment)](https://github.com/insightsengineering/thevalidatoR/pull/45#discussion_r756155629)

I added `additional_tlmgr_packages` parameter and just right now updated tests.

I'll wait then for your PR.

**Comment by dinakar29** [comment]
Merging this. Will release this as v1.1
